packages/net-stubbing/lib/server/route-matching.ts

```diff
@@ -121,3 +121,37 @@ export function getRouteForRequest (routes: BackendRoute[], req: CypressIncoming
     return _doesRouteMatch(route.routeMatcher, req)
   })
 }
+
+/**
+ * Is this a CORS preflight request that could be for an existing route?
+ * If there is a matching route with method = 'OPTIONS', returns false.
+ */
+export function matchesRoutePreflight (routes: BackendRoute[], req: CypressIncomingRequest) {
+  if (req.method !== 'OPTIONS' || !req.headers['access-control-request-method']) {
+    // not a valid preflight request
+    return false
+  }
+
+  let hasCorsOverride = false
+
+  const matchingRoutes = _.filter(routes, ({ routeMatcher }) => {
+    // omit headers from matching since preflight req will not send headers
+    const preflightMatcher = { ...routeMatcher }
```

Comment:
since this is shallow clone (almost spelled it clown), you could just use https://lodash.com/docs/4.17.15#omit

```js
const preflightMatcher = _.omit(routeMatcher, ['method', 'headers', 'auth'])
```